### PR TITLE
Pull request develop -> master

### PR DESCRIPTION
**Pull request by tannerilz**
Fixed bugs, stated by rsasov(where needed)!

### CODE REVIEW

File: src/main/java/gameengine/GameEngine.java [hunk 1]
```diff
@@ -6,7 +8,7 @@
  * It contains static final properties for implementing logic.
  * Methods for the fighting of the heroes and printing building a string with the flow of the game which
  * will be printed in main method.
- * GameEngine class contains main method.
+ * gameengine.GameEngine class contains main method.
```
Comment (vanyamihova-deltasource) [inline src/main/java/gameengine/GameEngine.java]: In Javadoc use in-line links {@link GameEngine}. Check this in google

File: src/main/java/gameengine/GameEngine.java [hunk 2]
```diff
@@ -111,15 +116,15 @@ public boolean isHeroDead(Hero hero) {
...
     public static void main(String[] args) {
-        //Creating object of the GameEngine
+        //Creating object of the gameengine.GameEngine
         GameEngine game = new GameEngine();
 
         //Two Hero objects for the fight
-        Hero heroOne = new Monk("Gosho");
-        Hero heroTwo = new Warrior("Pesho");
+        Hero heroOne = new Monk("Emir");
+        Hero heroTwo = new Warrior("Eren");
```
Comment (vanyamihova-deltasource) [inline src/main/java/gameengine/GameEngine.java]: Try with other names like `firstHero`, `secondHero` or even better to be `monk` and `warrior`.

File: src/main/java/gameengine/GameEngine.java [hunk 3]
```diff
@@ -27,6 +29,7 @@ public String fightSequence(Hero heroOne, Hero heroTwo) {
         String heroOneString = heroOne.getHeroName() + " the " + heroOne.getHeroType();
         String heroTwoString = heroTwo.getHeroName() + " the " + heroTwo.getHeroType();
         String damage;
+        String hasWon, hasDied;
```
Comment (vanyamihova-deltasource) [inline src/main/java/gameengine/GameEngine.java]: These variables are not so self-explanatory. It's a bad practice to start a String variable with a question word. Try to change them.

File: src/main/java/gameengine/GameEngine.java [hunk 4]
```diff
@@ -6,7 +8,7 @@
  * It contains static final properties for implementing logic.
  * Methods for the fighting of the heroes and printing building a string with the flow of the game which
  * will be printed in main method.
- * GameEngine class contains main method.
+ * gameengine.GameEngine class contains main method.
  * @author Taner
  */
 public class GameEngine {
```
Comment (vanyamihova-deltasource) [inline src/main/java/gameengine/GameEngine.java]: The method `public String fightSequence(Hero heroOne, Hero heroTwo) {` is with really long logic and it's hard to follow. Try to make several shorter methods with Single Responsibility Principle for every method.

File: src/test/java/testGameEngine.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import character.types.Assassin;
+import character.types.Knight;
+import character.types.Monk;
+import character.types.Warrior;
+import gameengine.GameEngine;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class testGameEngine {
```
Comment (vanyamihova-deltasource) [inline src/test/java/testGameEngine.java]: First letter of class name must be capitalized. :)

File: src/main/java/gameengine/GameEngine.java [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        }
+        return " dealt " + damage;
+    }
+
+    /**
+     * Method checking if the hero is dead after getting hit by an attack
+     * @param hero Object of hero type
+     * @return boolean type if the hero is dead
+     */
+    public boolean isHeroDead(Hero hero) {
```
Comment (nikolayivanov-deltasource) [inline src/main/java/gameengine/GameEngine.java]: @tannerilz This method has a confusing name. 'isHeroDead' implies that it checks if the hero is dead, but the method actually has a side effect - it updates the health points.
